### ISSUE TITLE
[Bug] Use Segmentation tools with Stack Viewport

### ISSUE THREAD

**Issue by 1isten** [cornerstonejs/cornerstone3D#712]
### Describe the Bug

This documentation link is broken:
https://www.cornerstonejs.org/docs/tutorials/basic-segmentation-tools/#read-more
<img width="786" alt="image" src="https://github.com/cornerstonejs/cornerstone3D/assets/11518941/3f521534-7d7e-4f56-b22f-a0f4804c5fb7">

https://www.cornerstonejs.org/live-examples/segmentationRendering.html
<img width="739" alt="image" src="https://github.com/cornerstonejs/cornerstone3D/assets/11518941/bee8423e-e119-418e-bde1-7095c7438ce3">

### Steps to Reproduce

Checkout above link

### The current behavior

The segmentation rendering example is broken, can't find any example of using segmentation tool for stack viewport.

### The expected behavior

Please give an example of using segmentation tool for stack viewport. In my case, I render single dicom file from my local filesystem (using stack viewport like [this example](https://www.cornerstonejs.org/live-examples/local.html)). But I can't figure out how to add segmentation tool then

### OS

macOS 13.3

### Node version

16.17.1

### Browser

Chrome 115.0.5790.114

**Comment by sedghi** [comment]
sorry about that it is wrong. I will edit it
Stack Viewport segmentation is not ready yet unfortunately but we are working on it

**Comment by 1isten** [comment]
@sedghi So currently we can only use segmentation in volume viewport?

If that's the case, may I ask another question (sorry I'm not familiar with cornerstone...)

Can I create a volume viewport to display just a single dicom file from my local filesystem?

**Comment by sedghi** [comment]
yes currently only in volume viewports. I haven't tried it it might work

**Comment by 1isten** [comment]
May I know if it is possible to load a single local dicom file such as this example: https://www.cornerstonejs.org/live-examples/local.html, but use volume viewport instead of stack viewport?

**Comment by sedghi** [comment]
You can look into our dicomImageLoader package for more infor on how to load. 

Nifti -> still in progress PR #696 

DICOM -> 

- Using WADO-RS and `dicomweb` standard https://www.dicomstandard.org/using/dicomweb and using open source pacs such as [orthanc](https://www.orthanc-server.com/) or [dcm4chee](https://www.dcm4che.org/). Most examples in the repo are this type which is our server in cloudfront (but you can achieve the same with other pacs)
 
see for instance example [here](https://github.com/cornerstonejs/cornerstone3D/blob/cfdb0b38e5cfee93f7c0d241a75498e08dcec6e8/packages/core/examples/stackAPI/index.ts)

```js
 const imageIds = await createImageIdsAndCacheMetaData({
    StudyInstanceUID:
      '1.3.6.1.4.1.14519.5.2.1.7009.2403.334240657131972136850343327463',
    SeriesInstanceUID:
      '1.3.6.1.4.1.14519.5.2.1.7009.2403.226151125820845824875394858561',
    wadoRsRoot: 'https://d3t6nz73ql33tx.cloudfront.net/dicomweb',
  });
```

Then `imageIds` are the path to the data that is created by cornerstone3D. Read more about imageIds [here](https://www.cornerstonejs.org/docs/concepts/cornerstone-core/imageId)

- Using [WADO-URI](https://dicom.nema.org/dicom/2013/output/chtml/part18/sect_6.2.html), example [here](https://github.com/cornerstonejs/cornerstone3D/blob/10a3370b7f23084d1f2c55506079c17dea959659/packages/core/examples/wadouri/index.ts)  
- Using dicomfile itself either [hosted](https://github.com/cornerstonejs/cornerstone3D/blob/4aee28f3898c52f07167dac1cd4c74a3e0ed8c89/packages/dicomImageLoader/examples/dicomImageLoaderWADOURI/index.ts) or as a [drag and drop](https://github.com/cornerstonejs/cornerstone3D/blob/a01687ab925c469bf979d6f2089d2e8f31c28e75/packages/tools/examples/local/index.ts)

**Comment by sedghi** [comment]
> May I know if it is possible to load a single local dicom file such as this example: [cornerstonejs.org/live-examples/local.html](https://www.cornerstonejs.org/live-examples/local.html), but use volume viewport instead of stack viewport?

You can convert stack to volumes and vice versa if you need, look at this [example](https://github.com/cornerstonejs/cornerstone3D/blob/c721ecd0a2724fa04c01704a33239e68eac5d0f1/packages/tools/examples/stackToVolumeWithAnnotations/index.ts)

**Comment by JiaBob** [comment]
Thanks for you answers! It takes time to dive into them. Let me try.

**Comment by 1isten** [comment]
Now I've used the orthanc server to retrieve metadata and successfully created the volume viewport to render some dicoms

But I found that even if the dicomweb server is used, cornerstone's volume viewport still cannot render some dicoms (such as US dicom files) which missing below tags:

- 0020,0037 - ImageOrientationPatient
- 0020,0032 - ImagePositionPatient
- etc...

when those tags metadata is missing, creating volume viewport causes below error:
![image](https://github.com/cornerstonejs/cornerstone3D/assets/11518941/778db3f5-944f-4ee6-90a5-b8c7a2d2390e)

I think the error is raised because `imagePlaneModule` metadata is missing some required value:
![image](https://github.com/cornerstonejs/cornerstone3D/assets/11518941/eab58025-c66f-4e64-a2c1-e63d2d7c177e)


So does that mean the volume viewport cannot be used for such dicom files or do you have any other solutions?  @sedghi 

Btw, may we know when Stack Viewport segmentation would be ready? Thanks

**Comment by sedghi** [comment]
No you are correct, if that metadata is missing you can't reconstruct a volume. 
We are pushing on making the stack segmentation work soon  #666

**Comment by sedghi** [comment]
https://github.com/cornerstonejs/cornerstone3D/pull/670
